channeldb/waitingproof.go

```diff
@@ -212,7 +212,7 @@ func (p *WaitingProof) Encode(w io.Writer) error {
 		b[0] = 1
 	}
 
-	if _, err := w.Write(b[:]); err != nil {
+	if err := binary.Write(w, byteOrder, b[:]); err != nil {
```

Comment:
We should be able to pass `p.isRemote` directly into `binary.Write` here, eliminating the declaration of `var b [1]byte` all together.